openquake/hazardlib/gsim/base.py

```diff
@@ -597,6 +597,20 @@ def disaggregate_poe(self, sctx, rctx, dctx, imt, iml,
             for i, idx in enumerate(iml_bin_indices)
         ])
 
+    def get_imts(self):
+        """
+        :returns: set of IMT strings, obtained from the underlying coefficients
+        """
+        imts = set()
+        for attr in dir(self):
+            val = getattr(self, attr)
+            if isinstance(val, CoeffsTable):
+                for sa in val.sa_coeffs:
+                    imts.add(str(sa))
+                for imt in val.non_sa_coeffs:
+                    imts.add(str(imt))
```

Comment:
How does this operate for the interpolated coefficients. For example, if there are coefficients for Sa 3.0 and Sa 4.0 then any value between Sa 3.0 and 4.0 is supported? This might depend on how you want to use it in whatever function will eventually invoke this method